src/routes/conversation/+server.ts

```diff
@@ -49,14 +49,26 @@ export const POST: RequestHandler = async ({ locals, request }) => {
 	}
 
 	// Use the model preprompt if there is no conversation/preprompt in the request body
-	preprompt = preprompt === undefined ? model?.preprompt : preprompt;
+	const preprompt = await (async () => {
+		if (values.assistantId) {
+			const assistant = await collections.assistants.findOne({
+				_id: new ObjectId(values.assistantId),
+			});
+			return assistant?.preprompt;
+		} else if (values.preprompt) {
+			return values.preprompt;
+		} else {
+			return model?.preprompt;
+		}
```

Comment:
```suggestion
		} else {
			return values?.preprompt ?? model?.preprompt;
		}
```

haven't tested. But I think we can just simplify it

Reply:
good catch thanks!